web/src/lib/utils/tree-utils.ts

```diff
@@ -1,23 +1,58 @@
-export interface RecursiveObject {
+export type RecursiveObject = {
   [key: string]: RecursiveObject;
-}
+};
 
-export const normalizeTreePath = (path: string) => path.replace(/^\//, '').replace(/\/$/, '');
+export const normalizeTreePath = (path: string) => (path.at(-1) === '/' && path.length > 1 ? path.slice(0, -1) : path);
 
-export function buildTree(paths: string[]) {
-  const root: RecursiveObject = {};
+export const getPathParts = (path: string) => {
+  const parts = path.split('/');
+  if (path[0] === '/') {
+    parts[0] = '/';
+  }
 
-  paths.sort();
+  if (path.at(-1) === '/') {
+    parts.pop();
+  }
+
+  return parts;
+};
+
+export const joinPaths = (...paths: string[]) => {
+  const filtered = paths.filter((path) => path.length > 0);
+  if (filtered.length < 2) {
+    return filtered[0] || '';
+  }
+  return filtered.map((path) => (path === '/' ? '' : path)).join('/');
+};
+
+export const getParentPath = (path: string) => {
+  const normalized = normalizeTreePath(path);
+  const last = normalized.lastIndexOf('/');
+  return last === -1 ? '' : normalized.slice(0, last);
+};
+
+export const isLeaf = (tree: RecursiveObject) => {
+  for (const entry in tree) {
+    if (entry !== '\0') {
+      return false;
+    }
+  }
```

Comment:
What is this \0 magic symbol and why is it used everywhere

Reply:
It's used to differentiate folders that have assets right under them from folders that don't.

Reply:
It should probably be a constant with a descriptive name.

Reply:
Ah, good point!